src/static/riot/analytics/analytics.tag

```diff
@@ -0,0 +1,425 @@
+        }
+
+        function build_chart_data(data, day_resolution, csv_format) {
+            let chart_data = _.map(data, data_point => {
+                let d = new Date(data_point._datefield)
+                d.setDate(d.getDate() + 1)
+
+                return {
+                    x: d,
+                    y: data_point.count,
+                }
+            })
+
+            chart_data.sort(function(a,b) {
+                return a.x - b.x
+            })
+
+            return chart_data
+         }
+
+        function update_chart(chart, data, day_resolution) {
+            chart.data.datasets[0].data = build_chart_data(data, day_resolution, false)
+            chart.update()
+        }
+
+        self.update_analytics = function (start, end, time_unit) {
+            if (!end) {
+                end = datetime.local()
+            }
+
+            // Django's __range is exclusive of the end date, so it must be incremented by one day to include it.
+            let date_parameters = {
+                start_date: start.toISODate(),
+                end_date: end.plus({day: 1}).toISODate(),
+                time_unit: time_unit,
+            }
+
+            CODALAB.api.get_analytics(date_parameters)
+                .done(function (data) {
+                    console.log(data)
+                    let time_unit = data.time_unit === 'day'
+
+                    update_chart(self.competitionsChart, data.competitions_data, time_unit)
+                    update_chart(self.submissionsChart, data.submissions_data, time_unit)
+                    update_chart(self.usersChart, data.users_data, time_unit)
+
+                    self.competitions_data = data.competitions_data
+                    self.submissions_data = data.submissions_data
+                    self.users_data = data.users_data
+
+                    self.update({
+                        users_total: data.registered_user_count,
+                        competitions: data.competition_count,
+                        competitions_published: data.competitions_published_count,
+                        start_date_string: data.start_date,
+                        end_date_string: data.end_date,
+                        submissions_made: data.submissions_made_count,
+                    })
+                })
+                .fail(function (a, b, c) {
+                    toastr.error("Could not load analytics data...")
+                })
+        }
+
+
+        // Shortcut buttons
+        self.time_range_shortcut = function(unit_selection) {
+            self.end_date = datetime.local()
+
+            if (unit_selection === 'month') {
+                self.start_date = self.end_date.minus({months: 1})
+            } else if (unit_selection === 'week') {
+                self.start_date = self.end_date.minus({weeks: 1}).plus({day: 1})
+            } else if (unit_selection === 'year') {
+                self.start_date = self.end_date.minus({years: 1})
+            }
```

Comment:
something _like_ this should clean up this `if` logic.
```javascript
diffs = {
    month = {months: 1},
    week = {days: 6},
    year = {years: 1},
}
self.start_date = self.end_date.minus(diffs[unit_selection]}
```

Reply:
Done.